spec/rubocop/cop/style/empty_case_condition_spec.rb

```diff
@@ -194,5 +194,19 @@
 
       it_behaves_like 'detect/correct empty case, accept non-empty case'
     end
+
+    context 'When using `return` in `when` clause and ' \
+            'assigning the return value of `case`' do
+      it 'does not register an offense' do
+        expect_no_offenses(<<-RUBY.strip_indent)
+          v = case
+              when x.a
+                1
+              when x.b
+                return 2 if foo
```

Comment:
What if the `return` is in the `else` clause? 🤔

Reply:
I overlooked it 💦  I added test code and implementation for `else` clause.